### DISCUSSION TITLE
Warm-starting bound multipliers

### DISCUSSION THREAD

**Discussion by dpo** [coin-or/Ipopt#728] [General]
I’m a little confused as to how to warm start the multipliers associated with bound constraints. The [documentation](https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_bound_mult_init_method) states that the only two accepted values for `bound_mult_init_method` are `constant` and `mu-based`. Why then does `get_starting_point()` request $z_L$ and $z_U$? If I set $z_L$ and $z_U$ to specific (positive) values, I don’t see any difference in the initial dual feasibility compared to when I just set them to ones. Could you please explain what options are relevant here and how to set $z_L$ and $z_U$?

Thank you!

**Comment by svigerske** [comment]
`get_starting_point()` should request values for z_L and z_U only if option [warm_start_init_point](https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_warm_start_init_point) has been enabled. `bound_mult_init_method` should only matter if `warm_start_init_point` is off, and then does not care about values in z_L and z_U.

**Comment by dpo** [comment]
Ok, thank you. My situation is that I have fixed variables that I want to manage myself (instead of IPOPT eliminating them). The fixed variables are indicated by identical lower and upper bounds. I’m trying to provide a good initial estimate of the multipliers for those fixed variables by setting $z_L$ to the opposite of the negative part, and $z_U$ to the positive part. Is that the right way to do it with IPOPT? I’m a bit confused because the initial dual feasibility reported by IPOPT is the same whether I warm start the multipliers or not.

**Reply by svigerske** [reply]
So you set option [fixed_variable_treatment](https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_fixed_variable_treatment) to `make_constraint`? It looks like in this case, the warm-start values for z_L and z_U are ignored by Ipopt and instead 0 is used:

https://github.com/coin-or/Ipopt/blob/a8e153f3d13a7c10980f60d5eec29c883df03e82/src/Interfaces/IpTNLPAdapter.cpp#L1618-L1623

If you are in this case, then it may be worth to do this todo.

**Comment by dpo** [comment]
Yes, I set `fixed_variable_treatment = make_constraint`. The code you point to explains the behavior I’m observing. Are you seeing clearly how it should be modified to use info from $z_L$ and $z_U$? Would that be a large change?

**Reply by svigerske** [reply]
Sorry for the delay. I hope this is fixed now on stable/3.14 (change 363143b).
